ui/src/app/workflows/components/workflow-details/workflow-details.tsx

```diff
@@ -255,6 +255,8 @@ export const WorkflowDetails = ({history, location, match}: RouteComponentProps<
             e => {
                 if (e.type === 'DELETED') {
                     setError(new Error('Workflow deleted'));
+                } else if (hasArtifactGCError(e.object.status.conditions)) {
+                    setError(new Error('Artifact garbage collection failed'));
```

Comment:
`conditions` may also include MetricsError, SubmissionError, SpecError, etc. Is the artifact GC error more special than other errors so that we only need to report the former?

Reply:
There may be a different place to surface this error, this is the implementation I went with to start since Alex specified this error should show the same way we show when an artifact is deleted.

Reply:
If it's not the right place, where is that different place to surface this error?

Reply:
I will have to surface this error in a different block of code. Currently, this implementation will keep the workflow diagram from appearing when the error appears which is expected for a deleted workflow but not failed artifact garbage collection.